spring-cloud-sleuth-instrumentation/src/test/java/org/springframework/cloud/sleuth/instrument/web/client/feign/FeignContextBeanPostProcessorTest.java

```diff
@@ -0,0 +1,21 @@
+package org.springframework.cloud.sleuth.instrument.web.client.feign;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.BeanFactory;
+import org.springframework.cloud.openfeign.FeignContext;
+import org.springframework.context.ApplicationContext;
+
+
+class FeignContextBeanPostProcessorTest {
+
+    @Test
+    void should_have_parent_after_post_processed() {
+        FeignContextBeanPostProcessor feignContextBeanPostProcessor = new FeignContextBeanPostProcessor(Mockito.mock(BeanFactory.class));
+        FeignContext feignContext = new FeignContext();
+        feignContext.setApplicationContext(Mockito.mock(ApplicationContext.class));
+        TraceFeignContext traceFeignContext = (TraceFeignContext) feignContextBeanPostProcessor.postProcessAfterInitialization(feignContext, "test");
+        Assertions.assertNotNull(traceFeignContext.getParent());
```

Comment:
Please use assertj assertions instead of junit ones. Also I understand that we expect the parent to be same as the one from line 17, right? Maybe we should assert on that?